examples/adr-example.cc

```diff
@@ -269,16 +269,31 @@ main(int argc, char* argv[])
     // Create NS
     ////////////
 
-    NodeContainer networkServers;
-    networkServers.Create(1);
+    Ptr<Node> networkServer = CreateObject<Node>();
+
+    // PointToPoint links between gateways and server
+    PointToPointHelper p2p;
+    p2p.SetDeviceAttribute("DataRate", StringValue("5Mbps"));
+    p2p.SetChannelAttribute("Delay", StringValue("2ms"));
+    // Store NS app registration details for later
+    P2PGwRegistration_t gwRegistration;
+    for (auto gw = gateways.Begin(); gw != gateways.End(); ++gw)
+    {
+        auto container = p2p.Install(networkServer, *gw);
+        auto serverP2PNetDev = DynamicCast<PointToPointNetDevice>(container.Get(0));
+        gwRegistration.emplace_back(serverP2PNetDev, *gw);
+    }
 
     // Install the NetworkServer application on the network server
     NetworkServerHelper networkServerHelper;
-    networkServerHelper.SetGateways(gateways);
     networkServerHelper.SetEndDevices(endDevices);
     networkServerHelper.EnableAdr(adrEnabled);
     networkServerHelper.SetAdr(adrType);
-    networkServerHelper.Install(networkServers);
+    for (const auto& [serverP2PNetDev, gwNode] : gwRegistration)
+    {
+        networkServerHelper.AddGatewayP2P(serverP2PNetDev, gwNode);
+    }
```

Comment:
I was thinking about directly passing a `P2PGwRegistration_t` collection to `AddGatewayP2P` and to move this loop inside the function. It cleans up the examples a little bit.

Reply:
I am in favor of this, even though I do not consider it blocking for this MR